pages/testing.tsx

```diff
@@ -47,11 +53,59 @@ const Testing = () => {
     web3Provider ? web3Provider.getSigner() : null
   );
 
+  const [contractState, setContractState] = useState<ContractState>(
+    ContractState.Open
+  );
+  const [contractExpirationTime, setContractExpirationTime] =
+    useState<string>("");
+
+  const [currentTime, setCurrentTime] = useState<string>("");
+  const [showSettle, setShowSettle] = useState(false);
+
+  useEffect(() => {
+    if (
+      currentTime &&
+      contractExpirationTime &&
+      currentTime > contractExpirationTime
+    ) {
+      setShowSettle(true);
+    }
+  }, [contractExpirationTime, currentTime, contractState]);
   // Get contract data and set values.
   useEffect(() => {
     if (web3Provider && !lspContract) {
       const signer = web3Provider.getSigner();
       const contract = createLSPContractInstance(signer, KNOWN_LSP_ADDRESS);
+      contract.contractState().then(async (cs: ContractState) => {
```

Comment:
Should we have this useeffect poll at all? Right now, it looks like it only runs on network change or startup.

Reply:
Yeah I think so. Keep in mind the web3provider will be defined *after* the user connects their wallet above -- or at least some of these values.

So there will be some queries we will make that might change once this view integrates with Onboard.